tools/dust/mutate.nim

```diff
@@ -0,0 +1,71 @@
+import
+  compiler / modules / modulegraphs,
+  compiler / ast / [ lineinfos, renderer, ast, astalgo, ]
+
+import spec
+import hashing
+
+proc node(k: TNodeKind): PNode = PNode(kind: k)
+
+type
+  Transformer = proc (n: PNode): PNode
+
+proc transform*(n: PNode; fn: Transformer): PNode =
+  assert not n.isNil
+  result = fn(n)
+  if result.isNil:
+    result = shallowCopy n
+    for i, child in pairs(n):
+      result.sons[i] = transform(child, fn)
+
+proc rebuild*(n: PNode; fn: Transformer): PNode =
+  assert not n.isNil
+  result = fn(n)
+  if cast[uint](result) == cast[uint](n):
```

Comment:
```suggestion
  if result == n:
```
The equality operator for `PNode` is not overloaded, so this should do the same.